### PR TITLE
added saveConfiguration boolean to openGraphicsSettingsDialog()

### PR DESCRIPTION
**Pull request by glitchhopcore**
implemented it in a similar fashion to how openAudioSettingsDialog or how openInterfaceSettingsDialog has been implemented

refering to issue number #8489

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @glitchhopcore , I think this function should just return a boolean value which parent function will consider whether to save or not configuration file. Please check from where this function is being called. Also, as a requirement for the original issue we have multiple windows, not only graphics.

**Comment by glitchhopcore** [comment]
Hi @zenseii, I will make sure to fix all the code formatting issues soon. Besides that, @ihhub If I understand this correctly, in this case parent function would refer to openGameSettings() [for Audio and Graphics] and showSystemOptionsDialog() [for Interface Setting]? I would make the dialog functions to return a boolean based on if any of the settings requiring saving configuration file is called and inside the parent function just OR those flags and save the configuration file at once?

**Comment by ihhub** [comment]
@glitchhopcore , the idea is to save configuration file only once. Right now parent and child windows have independent configuration file saving. It should be saved only on the level of parent dialog.

File: src/fheroes2/dialog/dialog_graphics_settings.cpp [hunk 1]
```diff
@@ -233,29 +235,31 @@ namespace fheroes2
...
                 windowType = SelectedWindow::Configuration;
                 break;
             case SelectedWindow::SystemInfo:
                 conf.setSystemInfo( !conf.isSystemInfoEnabled() );
-                conf.Save( Settings::configFileName );
+                saveConfiguration = true;
                 windowType = SelectedWindow::Configuration;
                 break;
             default:
                 return;
```
[the remark this answers is not in the document]
Comment (glitchhopcore) [reply src/fheroes2/dialog/dialog_graphics_settings.cpp:259]: It needs to return a value for the openEditorSettings as well, making that change

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@glitchhopcore , thank you so much for your contribution to the project!

**Pull request merged**